### PR TITLE
:wrench: Add toggle

### PR DESCRIPTION
**Pull request by GestaltEngine**
:wrench: Add ToggleMenu D-Bud method. 
Best used in combination with xbindkeys to open/close menus with a single mouse button.

### CODE REVIEW

**Comment by Schneegans** [comment]
Thanks a lot, I'll test it tomorrow! And sorry that there was still the `develop` branch around. I simplified Fly-Pie's branching scheme with the latest release and totally forgot to delete it :wink:

File: src/extension/Daemon.js [hunk 1]
```diff
@@ -346,6 +346,16 @@ var Daemon = class Daemon {
 
     return 0;
   }
+  
+  // Opens a menu with that name if there are currently none open. Closes the currently
```
Comment (Schneegans) [inline src/extension/Daemon.js:350]: I think `clang-format` fails due to this trailing space here:

```suggestion
  // Opens a menu with that name if there are currently none open. Closes the currently
```

File: src/extension/Daemon.js [hunk 2]
```diff
@@ -346,6 +346,16 @@ var Daemon = class Daemon {
...
   }
+  
+  // Opens a menu with that name if there are currently none open. Closes the currently 
+  // open menu otherwise.   
+  ToggleMenu(name) {
+    if (this._menu.getID() == null) {
+      return ShowMenu(name);
+    } else {
+      return CancelMenu();
+    }
```
Comment (Schneegans) [inline src/extension/Daemon.js:357]: Here are two `this` missing:

```suggestion
    if (this._menu.getID() == null) {
      return this.ShowMenu(name);
    } else {
      return this.CancelMenu();
    }
```

**Review by Schneegans — CHANGES_REQUESTED** [review]
Again, thank you very much! I think that this is a small but very useful contribution. I just added two comments which you could directly commit, i think.

Furthermore, it would be cool if you could add a small example of this new method to the [documentation](https://github.com/Schneegans/Fly-Pie/blob/main/docs/dbus-interface.md#opening-menus-configured-with-the-menu-editor). Maybe right after the `ShowMenuAt` example.

**Comment by GestaltEngine** [comment]
This should be it. Clang-format didn't like the trailing spaces indeed. 
Added a couple of lines into documentation. Maybe they would be better placed after the `CloseMenu` example, considering that it inherits the functionality from both `ShowMenu` and `CloseMenu`?

**Comment by GestaltEngine** [comment]
Sorry such a small change is taking so long. I will be more thorough with future PRs.

**Review by Schneegans — APPROVED** [review]

**Comment by Schneegans** [comment]
Thanks again, it looks good now! I just realized that one problem is currently that a script couldn't tell if a menu was closed or opened by the method. So after I merged it, I'll make a tiny tweak so that the method returns a negative number if it canceled a menu. 

> I will be more thorough with future PRs.

I don't mind how long or how many "revert" commits it takes as long as the result is a cool contribution! So I am definitely looking forward to to your future PRs :wink:

Thank you once more!

**Pull request merged**